mozci/data/sources/activedata/__init__.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+from argparse import Namespace
+from collections import defaultdict
+from pathlib import Path
+
+import adr
+from loguru import logger
+
+from mozci.data.base import DataSource
+
+here = Path(__file__).parent.resolve()
+adr.sources.load_source(here)
+
+
+class ActiveDataSource(DataSource):
+    """Uses 'adr' to query ActiveData."""
+
+    name = "adr"
+    supported_contracts = (
+        "push_tasks",
+        "push_test_groups",
+    )
+
+    @classmethod
+    def normalize(cls, result):
+        if "header" in result:
+            result["data"] = [
+                {
+                    field: value
+                    for field, value in zip(result["header"], entry)
+                    if value is not None
+                }
+                for entry in result["data"]
+            ]
+
+        items = []
+        retries = defaultdict(int)
+
+        for item in result["data"]:
+            if "id" not in item:
+                logger.trace(f"Skipping {item} because of missing id.")
+                continue
+
+            task_id = item["id"]
+
+            # If a task is re-run, use the data from the last run.
```

Comment:
Maybe this logic should not be in the datasource implementation but in the push module

Reply:
So it doesn't have to be repeated in all datasources

Reply:
Maybe, I think I'd like to leave it here for now though, as I'm not sure that all datasources will even have the concept of a re-run.

In general I think we should be fairly fluid with the contracts going forward (change them to suit the needs of our data sources).